pkg/operator/operator.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operator
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	nodeobservabilityv1alpha1 "github.com/openshift/node-observability-operator/api/v1alpha1"
+	nodeobservabilityv1alpha2 "github.com/openshift/node-observability-operator/api/v1alpha2"
+	operatorconfig "github.com/openshift/node-observability-operator/pkg/operator/config"
+	controller "github.com/openshift/node-observability-operator/pkg/operator/controller"
+	machineconfigcontroller "github.com/openshift/node-observability-operator/pkg/operator/controller/machineconfig"
+	nodeobservabilitycontroller "github.com/openshift/node-observability-operator/pkg/operator/controller/nodeobservability"
+	nodeobservabilityrun "github.com/openshift/node-observability-operator/pkg/operator/controller/nodeobservabilityrun"
+)
+
+var (
+	setupLog = ctrl.Log.WithName("operator")
+)
+
+// Operator hold the manager resource.
+// for the nodeobservability opreator.
+type Operator struct {
+	manager manager.Manager
+}
+
+// New creates a new operator from cliCfg and opCfg.
+func New(cliCfg *rest.Config, opCfg *operatorconfig.Config) (*Operator, error) {
+	token, err := ioutil.ReadFile(opCfg.TokenFile)
+	if err != nil {
+		setupLog.Error(err, "unable to read serviceaccount token")
+		os.Exit(1)
```

Comment:
```suggestion
		return nil, fmt.Errorf("unable to read serviceaccount token: %w", err)
```
And the same everywhere we have `Exit()`, it's not `main.go` anymore.